cardano-config/src/Cardano/Config/Git/RevFromGit.hs

```diff
@@ -0,0 +1,25 @@
+module Cardano.Config.Git.RevFromGit (
+      gitRevFromGit
+    ) where
+
+import           Cardano.Prelude
+import           Prelude (String)
+
+import qualified Language.Haskell.TH as TH
+import           System.IO.Error (ioeGetErrorType, isDoesNotExistErrorType)
+import           System.Process (readProcessWithExitCode)
+
+-- | Git revision found by running git rev-parse. If git could not be
+-- executed, then this will be an empty string.
+gitRevFromGit :: TH.Q TH.Exp
+gitRevFromGit = TH.LitE . TH.StringL <$> TH.runIO runGitRevParse
+    where
+        runGitRevParse :: IO String
+        runGitRevParse = handleJust missingGit (const $ pure "") $ do
+            (exitCode, output, _) <-
+                readProcessWithExitCode "git" ["rev-parse", "--verify", "HEAD"] ""
+            pure $ case exitCode of
+                ExitSuccess -> output
+                _           -> ""
+
+        missingGit e = if isDoesNotExistErrorType (ioeGetErrorType e) then Just () else Nothing
```

Comment:
Why are we reverting this and not incorporate the fixes from `cardano-config`? This code here is clearly broken with some `process` libraries. If `git` is not found, this does not necessarily result in a `isDoesNotExistErrorType` but some other exception that is thrown. It's fairly annoying that the process lirbary got changed to support spawn, and now has a different behaviour. But this code is sadly broken as it is.

Reply:
```
            # src/Cardano/Config/Git/Rev.hs:33:35: error:
            #     • Exception when trying to run compile-time code:
            #         git: readCreateProcessWithExitCode: posix_spawn_file_actions_adddup2(child_end): invalid argument (Invalid argument)
            #       Code: gitRevFromGit
            #     • In the untyped splice: $(gitRevFromGit)
            #    |
            # 33 |         fromGit = T.strip (T.pack $(gitRevFromGit))
            #    |
            ```

Reply:
(And similar exceptions...)

Reply:
Copied the code fix across from https://github.com/input-output-hk/cardano-config/pull/4